docs/_chapters/230-manifest-annotations.md

```diff
@@ -1,136 +1,286 @@
+The invisible link created between user code and the indirect requirement is a powerful mechanism that enables automatic validation of a bundle closure.
 
-	@RequireCapability(
-	  ns        = “osgi.extender”, 
-	  filter    = "(&(osgi.extender=osgi.enroute.configurer)${frange;1.2.3})",
-	  effective = "active")
-	public class Peggy { ... }
+### Arbitrary Manifest Headers
 
-This generates the following manifest headers (though the actual header can be larger since it can contain additional requirements):
+_Bundle annotations_ aren't just about package versioning or requirements and capabilities. They are about lifting *metadata* out of our code to avoid, among other things, error prone duplication of information. A common example is the bundle activator. Bundle Activators are require to be described in a manifest header. This association is not visible to refactoring tools and as such can easily end up out of sync.
 
-	Require-Capability: osgi.extender;filter:='(&(osgi.extender=osgi.enroute.configurer`)
-		(&(version>=1.2.3)(!(version>=2.0.0))))';effective:=active
+The [`@Header`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Header) annotation exists to address this problem.
 
-It is also possible to create a customized annotation:
+```java
+package com.acme;
 
-	@RequireCapability(
-	  ns        = “osgi.extender”, 
-	  filter    = "(&(osgi.extender=osgi.enroute.configurer)${frange;1.2.3})",
-	  effective = "active")
-	@Retention(RetentionPolicy)
-	@Target( ElementType.TYPE )
-	public @interface RequireConfigurer {}
+@Header(name = Constants.BUNDLE_ACTIVATOR, value = "${@class}")
+public Activator implements BundleActivator { ... }
+```
 
-This by itself does not create a clause in the manifest header. To make this happen we need to annotate a type:
+results in the manifest header:
 
-	@RequireConfigurer
-	public class Peggy { ... }
+`Bundle-Activator: com.acme.Activator`
 
-Now the same header will be generated.
 
-## Bundle License
+### Macros
 
-The `@BundleLicense` annotation creates entries in the Bundle-License header. The annotation has the following fields:
+You'll note the string `${@class}` used in the above example. String fields in *bundle annotations* are processed through bnd's macro processor. This macro processor provides access to all default and builder macros. *More info on bnd macros can be found in the [macros chapter](/chapters/850-macros.html).* 
 
-* `name` – The name of the license, should preferably the URI to the corresponding [Open Source Initiative][2] page about this license.
-* `description` – A short description of the license.
-* `url` – The URL to further information about the license
+Bnd also provides access to certain key properties of the current processing state.
 
-A number of licenses have gotten their own custom annotation:
+- ***@class*** - gives the fully qualified name of the class where the annotation hierarchy began
```

Comment:
It is the annotated class. I am not entirely sure what "the class where the annotation hierarchy began" means.